agent/agent.go

```diff
@@ -2872,7 +2886,12 @@ func (a *Agent) loadServices(conf *config.RuntimeConfig) error {
 		ns.Connect.SidecarService = nil
 
 		if err := a.addServiceLocked(ns, chkTypes, false, service.Token, false, ConfigSourceLocal); err != nil {
-			return fmt.Errorf("Failed to register service %q: %v", service.Name, err)
+			if a.config.IgnoreBadCheckFiles == false {
```

Comment:
Weird nitpick, instead of checking if the flag is set to false, is there anyway we could rework the code to do something if it is true?

Reply:
Yeah, I see that here too. My mind working the defaults and deviations from defaults... I can refactor this.

Reply:
That would be wonderful, thank you!